arc/recommendations/generic_analyzer.py

```
@@ -0,0 +1,437 @@
+        if any(word in prompt_lower for word in ["search", "find", "lookup", "query"]):
+            return "search_task"
+        elif any(word in prompt_lower for word in ["calculate", "compute", "math", "number"]):
+            return "calculation_task"
+        elif any(word in prompt_lower for word in ["analyze", "parse", "process", "extract"]):
+            return "analysis_task"
+        elif any(word in prompt_lower for word in ["write", "create", "generate", "compose"]):
+            return "generation_task"
+        else:
+            return "general_task"
+    
+    def _extract_execution_insights(self, trajectories: List[Dict]) -> Dict[str, Any]:
```

Ruff (0.11.9)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)